### ISSUE TITLE
PC-BSD: Use of uninitialized value $apicall in ...

### ISSUE THREAD

**Issue by abacefbeca** [HikariKnight/rsu-client#126]
I'm using PC-BSD 10.3 64-bit. I downloaded and ran the [32-bit Linux installer](https://github.com/HikariKnight/rsu-client/releases/download/v4.3.4/RSU-linux-installer.run.zip). The 64-bit installer didn't do anything. The 32-bit installer identified the operating system as CentOS. That was expected. At first the install-desktop-icons script didn't work. I fixed that by using the commands:
`sudo ln -s /usr/local/bin/perl /usr/bin/perl`
`sudo ln -s /usr/local/bin/xterm /usr/bin/x-terminal-emulator`
Tried to run the runescape script but received various errors and warnings because of missing dependencies. Solved those issues with the following command:
```
sudo pkg install openjdk p5-Wx p5-IPC-Cmd p5-Locale-Maketext-Simple p5-Module-Load-Conditional p5-Params-Check p5-Archive-Extract p5-Config-IniFiles p5-List-MoreUtils p5-IO-Stringy p5-Config-IniFiles
```
I got OpenJDK version 1.7.0_101. In the list of dependencies shown while running the RSU installer there is a typo on line 28: the C and P in ICP need to switch places. The command above fixed several error messages but there are some error messages I haven't figured out:
```
[User1@pcbsd-6326] ~/runescape% ./runescape                                     WARNING: Running in fallback mode!
rsu-queries will use the system perl so some API calls
may not work until you install certain modules
Fontconfig warning: "local.conf", line 1093: saw number, expected matrix
Use of uninitialized value $apicall in substitution (s///) at /usr/home/User1/runescape/rsu/apiquery.pm line 43.
Use of uninitialized value $apicall in concatenation (.) or string at /usr/home/User1/runescape/rsu/apiquery.pm line 46.
syntax error at (eval 45) line 1, at EOF
	...caught at /usr/home/User1/runescape/rsu/apiquery.pm line 46.
```
I think the Fontconfig warning is just a general bug unrelated to my issue. The errors I'm concerned about are the "Use of uninitialized value ..." and syntax errors. Am I still missing a package? I was able to start OldSchool RS by running
`[User1@pcbsd-6326] ~/runescape% ./runescape client.launch.runescape --prmfile=oldschool.prm`
after editing /usr/home/User1/runescape/rsu/apiquery.pm and pointing it to the java binary. However, I'd like to be able to use the GUI menus for the RSU launcher.

**Comment by abacefbeca** [comment]
@HikariKnight

**Comment by abacefbeca** [comment]
I also tried installing `p5-File-Temp p5-constant p5-File-Copy-Recursive p5-Encode p5-Carp p5-FindBin-libs p5-Exporter p5-HTTP-Tiny p5-IO-Handle-Util p5-File-Path p5-Encode p5-JSON-PP` but that didn't fix the errors.

**Comment by aimeec1995** [comment]
I think it's time to take down the claims that this is supported on bsd/solaris.

**Comment by HikariKnight** [comment]
yes i agree with @aimeec1995 as something have happened since i last tested it under bsd where it ran fine, and as i do not have time to test and see what have changed i will have to remove the statement for support for it and set bsd/solaris and non x86 and x86_64 architectures as untested.

**Comment by abacefbeca** [comment]
I don't know Perl but have some experience with programming, can use Google, and the script comments help.

On line 41 in apiquery.pm $ARGV[0] doesn't contain what it needs to have. If I type `runescape` in Terminal I get the errors I mentioned in my first post. If I type `runescape %U` as shown by the properties of the desktop icon then the errors in my first post don't happen but I get the following error:
```
syntax error at (eval 45) line 1, near "use %U"
Global symbol "%U" requires explicit package name at (eval 45) line 1.
	...caught at /usr/home/User1/runescape/rsu/apiquery.pm line 46.
```
Line 46 of apiquery.pm needs $apicall to be a module name but it isn't. Seems like the script doesn't work if the operating system is unrecognized (fallbackmode) and $ARGV[0] is empty or contains something strange.

If I do `./runescape/rsu/rsu-query client.launch.launcher` in Terminal from home folder then the launcher starts. Also works if I do `./runescape/runescape client.launch.launcher`

**Comment by aimeec1995** [comment]
hey abacefbeca  how can i contact you?

**Comment by abacefbeca** [comment]
@aimeec1995 
Looks like neither of us wants to share our email address in the public profile of our Github account and Github removed the ability to send private messages so I tried contacting you on the FreeBSD forum.
